src/ucp/core/ucp_listener.c

```diff
@@ -218,6 +217,62 @@ static void ucp_listener_close_ifaces(ucp_listener_h listener)
     ucs_free(listener->wifaces);
 }
 
+static void
+ucp_listener_conn_request_cb(uct_listener_h listener, void *arg,
+                             const char *local_dev_name,
+                             uct_conn_request_h conn_request,
+                             const uct_cm_remote_data_t *remote_data)
+{
+    ucp_listener_h ucp_listener = arg;
+    uct_worker_cb_id_t prog_id  = UCS_CALLBACKQ_ID_NULL;
+    ucp_conn_request_h ucp_conn_request;
+    ucs_status_t status;
+
+    ucp_conn_request = ucs_malloc(ucs_offsetof(ucp_conn_request_t, client_data) +
+                                  remote_data->conn_priv_data_length,
+                                  "ucp_conn_request_h");
+    if (ucp_conn_request == NULL) {
+        ucs_error("failed to allocate connect request, rejecting connection request %p on TL listener %p, reason %s",
+                  conn_request, listener, ucs_status_string(UCS_ERR_NO_MEMORY));
+        goto err_reject;
+    }
+
+    ucp_conn_request->remote_dev_addr = ucs_malloc(remote_data->dev_addr_length,
+                                                   "remote device address");
+    if (ucp_conn_request->remote_dev_addr == NULL) {
+        ucs_error("failed to allocate device address, rejecting connection request %p on TL listener %p, reason %s",
+                  conn_request, listener, ucs_status_string(UCS_ERR_NO_MEMORY));
+        goto err_free_ucp_conn_request;
+    }
+
+    ucp_conn_request->listener     = ucp_listener;
+    ucp_conn_request->uct.listener = listener;
+    ucp_conn_request->uct_req      = conn_request;
+    memcpy(ucp_conn_request->remote_dev_addr, remote_data->dev_addr,
+           remote_data->dev_addr_length);
+    memcpy(&ucp_conn_request->client_data, remote_data->conn_priv_data,
```

Comment:
where do we use this client_data?

Reply:
it contains remote address, we use it at EP creation phase